### PR TITLE
Linting: Enable `PLR6301`, # could be a function, class method or static method

### PR DESCRIPTION
**Pull request by samet-akcay**
## 📝 Description

- Enable `PLR6301`, # could be a function, class method or static method

## ✨ Changes

Select what type of change your PR is:

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🔨 Refactor (non-breaking change which refactors the code base)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update
- [ ] 🔒 Security update

## ✅ Checklist

Before you submit your pull request, please make sure you have completed the following steps:

- [ ] 📋 I have summarized my changes in the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) and followed the guidelines for my type of change (skip for minor changes, documentation updates, and test enhancements).
- [ ] 📚 I have made the necessary updates to the documentation (if applicable).
- [ ] 🧪 I have written tests that support my changes and prove that my fix is effective or my feature works (if applicable).

For more information about code review checklists, see the [Code Review Checklist](https://github.com/openvinotoolkit/anomalib/blob/main/docs/source/markdown/guides/developer/code_review_checklist.md).

### CODE REVIEW

File: tests/unit/engine/test_setup_transform.py [hunk 1]
```diff
@@ -41,17 +41,19 @@ def __init__(self) -> None:
         super().__init__()
         self.model = torch.nn.Linear(10, 10)
 
-    def configure_transforms(self, image_size: tuple[int, int] | None = None) -> Transform:
+    def configure_transforms(self, image_size: tuple[int, int] | None = None) -> Transform:  # noqa: PLR6301
```
Comment (ashwinvaidya17) [inline tests/unit/engine/test_setup_transform.py:44]: Why can't we make this a static method?

**Review by ashwinvaidya17 — COMMENTED** [review]
Thanks for the efforts! I have a few questions.

What happens if instead of ignoring the rule, we convert methods like `configure_transforms`, `on_train_start` (and other lightning methods), `forward`, and `backward` to static?

File: tests/unit/engine/test_setup_transform.py (hunk 1 above)
Comment (samet-akcay) [reply tests/unit/engine/test_setup_transform.py:44]: I was not sure if any of the models would need `self` in the transforms. If you think it is safe to make it static method, I could change that

Comment (samet-akcay) [reply tests/unit/engine/test_setup_transform.py]: I double checked, and converted the child class implementations. I left the some of the parent classes as is though.

**Pull request merged**